src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_sysfs.c

```diff
@@ -765,6 +765,18 @@ static ssize_t host_mem_size_show(struct device *dev,
 }
 static DEVICE_ATTR_RO(host_mem_size);
 
+static ssize_t versal_show(struct device *dev,
+	struct device_attribute *attr, char *buf)
+{
+	struct xocl_dev *xdev = dev_get_drvdata(dev);
+	uint32_t val = 0;
+
+	val = (XOCL_DSA_IS_VERSAL_ES3(XDEV(xdev))) ? 1 : 0;
```

Comment:
This is slightly different from the implementation of the mgmtpf. Are they reporting the same information? If so they should both be the same. Maybe changing the mgmtpf portion to use XOCL_DSA_IS_VERSAL_ES3 instead of the bit comparison?

Reply:
Oh.. I didn't take a close look at the mgmt implementation. It should use macro which is more readable. I will update it.

Reply:
@dbenusov-xilinx The XOCL_DSA_IS_VERSAL_ES3 macro is only for userpf. There is no similar macro defined in mgmtpf. Actually, the coding style in mgmtpf side is always use bit comparison... I will not make that change.

On the userpf side, XOCL_DSA_IS_VERSAL_ES3 is for the discovery and newer versal shell. That is the target shell for PS kernel. Unless we want to run PS kernel on legacy versal shell. Otherwise, I think current solution is sufficient.